rfd/0167-auto-updates-change-proposal.md

```diff
@@ -0,0 +1,172 @@
+https://updates.releases.teleport.dev/v1/stable/cloud/v15/version -> v15.1.1
+```
+
+### Proxy Version Server
+The Teleport proxies now support a version endpoint and serve the latest compatible agent version. The Teleport Cloud proxies are configured to forward version requests to the appropriate upstream major version channel.
+```
+# Forwards to https://updates.releases.teleport.dev/v1/stable/cloud/v14/version
+https://platform.teleport.sh:443/v1/webapi/automaticupgrades/channel/default/version -> v14.3.6
+```
+
+### Teleport Updater Version Endpoint
+The Teleport updater is now configured to request an agent version from the proxy instead of the global version server.
+
+## Overview
+With the above changes, this is now what the automatic updates architecture looks like.
+
+### Version Channels
+Teleport Cloud maintains major version channels along with the deprecated global major version channel. Teleport proxies support a version endpoint that serves the latest compatible version of Teleport. The proxies request the latest available version from the appropriate upstream version channel. New Teleport updaters (v14.3.1+), request the target update version from the Teleport proxy. Old Teleport updaters (<v14.3.1) continue to request the target update version from the global version channel.
+
+![version-channels](assets/0167-auto-updates-change-proposal/version-channels.png)
+
+### Publishing
+Teleport maintains a stable/cloud channel for APT and RPM packages. After updating Teleport Cloud tenants to vX.X.X, Teleport Cloud deploys a GitHub workflow to publish the necessary packages. The workflow publishes vX.X.X teleport-ent and teleport-ent-updater packages to the stable/cloud repositories, and then updates the version channels.
+
+Teleport also maintains a Helm repository. The teleport/teleport-kube-agent Helm chart is published regularly with the OSS release of Teleport.
+
+![publishing](assets/0167-auto-updates-change-proposal/publishing.png)
+
+### Systemd Deployments
+The documentation for deploying Teleport on a Linux server instruct users to install two packages: teleport-ent and teleport-ent-updater. The teleport-ent-updater package installs the teleport-upgrade tool and starts a teleport-upgrade.timer/service to periodically check for available updates. The teleport-upgrade tool invokes the package manager to handle teleport-ent package updates.
+
+![publishing](assets/0167-auto-updates-change-proposal/systemd-deployment.png)
+
+### Kubernetes Deployments
+Kubernetes deployments of Teleport are installed using Helm. The teleport/teleport-kube-agent Helm release deploys the teleport-agent resource and a separate teleport-agent-updater resource at the same version. The teleport-agent-updater periodically checks for available updates and updates the teleport-agent resource image version.
+
+![publishing](assets/0167-auto-updates-change-proposal/kube-deployment.png)
+
+## Current Limitations
+
```

Comment:
I would like to add another current limitation: the automatic update process only covers the agents. Not the integrations.

Since v15 we have Teleport Operator users against Cloud instances, we already had users self-hosting plugins ([see slack link](https://goteleport.slack.com/archives/C0582MBSMHN/p1710343435263599?thread_ts=1710342970.418989&cid=C0582MBSMHN)).

I suspect that as Teleport Cloud grows and operator adoption increases, similar issues will arise. I think the proposed solution should be designed to be extended in the future to cover the integrations, at least self-hosted plugins and the operator.